### DISCUSSION TITLE
Texture fading on rescaling/resizing

### DISCUSSION THREAD

**Discussion by SanicBTW** [ppy/osu-framework#6278] [Q&A] [answered]
Hi everyone, I'm currently working on a project that uses sparrow spritesheets and I got the parsing working and all BUT I have a problem, when I try to resize a sprite which texture was made through the parsing, it ends up making this weird fading on the edges, I tried pretty much everything as setting the filtering mode or wrapping but nothing seems to be working, also I recently tried BufferedContainer but still did nothing, I am using a TextureRegion to define the texture from the spritesheet, I would love some alternatives or answers to why this happens

**Comment by peppy** [comment]
without any visuals to go by, it sounds like the corner pixels of each of your sprites in the spritesheet are sampling nearby transparent pixels due to a min/mag filter. you might get better results by setting the texture filtering mode to `nearest` (on your `TextureStore`), or by filling in nearby pixels with the expected edge colour.

again, hard to say without further details.

**Reply by SanicBTW** [reply]
My bad, I didn't have any visual at the moment when I was writing the post

![image](https://github.com/ppy/osu-framework/assets/58010536/ecd953c9-7695-47f1-b538-418e03151e0e) (using BufferedContainer) 

![image](https://github.com/ppy/osu-framework/assets/58010536/caba53df-6a8c-4657-a513-fad8cb8ffdea) (using scaling)

I'll look into what you suggested rq, thanks!

**Reply by smoogipoo** [reply]
Rather than changing the filter mode to nearest, you likely want to use `ClampToEdge` https://github.com/ppy/osu-framework/blob/004fae48aeff544b9253179185624aafd5894909/osu.Framework/Graphics/Textures/WrapMode.cs#L13-L16

E.g. `TextureStore.Get(..., WrapMode.ClampToEdge, WrapMode.ClampToEdge)`

**Reply by SanicBTW** [reply]
it still happens even when setting the wrap mode to ClampToEdge
![image](https://github.com/ppy/osu-framework/assets/58010536/a6c6d98b-d703-4ef7-940f-b005a1b6c31c)

**Reply by SanicBTW** [reply, marked as answer]
Alright so after messing around a lil bit more with BufferedContainers, I found out that setting the FrameBuffer Y Scale to 0 fixes the texture fading

FrameBuffer Scale Y 1
![image](https://github.com/ppy/osu-framework/assets/58010536/bbb7e0d8-df43-4588-9764-86d48f1c38b0) 

FrameBuffer Scale Y 0
![image](https://github.com/ppy/osu-framework/assets/58010536/f826065e-6ff5-4f26-887a-3cb964eed7fa)

![image](https://github.com/ppy/osu-framework/assets/58010536/36bbf326-17ab-4745-b2a1-87505ab52e4f)

I guess it works because the sprite doesn't find any nearby transparent pixels available on such a small texture size, anyways thanks for the help guys!

**Reply by smoogipoo** [reply]
Oh. Using buffered containers is a bit of an important detail. We haven't really paid much attention to those being filtered pixel-perfectly yet, but I'll just say that using a 0x scale sounds like a jank solution and I'd hope that we can do this better in the future. I have nothing better so if it works then I can't argue against that solution, though! 😄

**Reply by SanicBTW** [reply]
Yeah I was suprised that using a 0x scale would fix my issue, I feel like there should be another way of fixing it, but if it works I might just stay like this, I haven't really noticed any performance drops when moving to a BufferedContainer, everything is pretty fast, keep up the great job guys! I really like to see how the framework keeps improving everyday 🔥
